modyn/config/schema/pipeline.py

```diff
@@ -364,17 +362,16 @@ def validate_gpus(cls, value: int) -> int:
         return value
 
     @model_validator(mode="after")
-    @classmethod
-    def validate_num_samples_to_pass(cls, data: "TrainingConfig") -> "TrainingConfig":
-        if data.initial_model == "pretrained":
-            if not data.use_previous_model:
-                raise ValidationError(
+    def validate_pretrained(self) -> Self:
+        if self.initial_model == "pretrained":
+            if not self.use_previous_model:
+                raise ValueError(
                     "Cannot have use_previous_model == False and use a pretrained initial model."
                     "Initial model would get lost after first trigger."
                 )
-            if not data.initial_model_id:
-                raise ValidationError("Initial model set to pretrained, but no initial_model_id given")
-        return data
+            if not self.initial_model_id:
+                raise ValueError("Initial model set to pretrained, but no initial_model_id given")
```

Comment:
Also `ValidationError` should not used, see https://docs.pydantic.dev/latest/concepts/validators/#handling-errors-in-validators. Actually `ValidationError` even does not have a constructor with a string.